### PR TITLE
cosi deployment docs

### PR DESCRIPTION
**Pull request by tparikh**

### CODE REVIEW

File: docs/deployment-guide.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Deploying Container Object Storage Interface (COSI) On Kubernetes
+
+This document describes steps for Kubernetes administrators to setup Container Object Storage Interface (COSI) onto a Kubernetes cluster.
+
+## Overview
+
+There are four components that need to be deployed in Kubernetes to setup COSI.
```
Comment (wlan0) [inline docs/deployment-guide.md:7]: :+1

File: docs/deployment-guide.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+- CustomResourceDefinitions (CRDs)
+- Controller
+- Provisioner Sidecar
```
Comment (wlan0) [inline docs/deployment-guide.md:11]: - Driver 
- Sidecar for the driver

Generally they are run as a part of the same pod

File: docs/deployment-guide.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+- Node Adapter
+
```
Comment (wlan0) [inline docs/deployment-guide.md:13]: Here is the **_extremely_** quick start:

```sh
kubectl create -k ...
# this deploys the sample-driver. To replace the driver with your own driver, refer to [docs](#installing-driver)
kubectl create -k ...

```

File: docs/deployment-guide.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+- [BucketAccessRequest](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1979-object-storage-support#bucketaccessrequest), which is a user namespaced custom resource representing an object store user and an access policy defining the user’s relation to that storage.
+- [BucketAccessClass](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1979-object-storage-support#bucketaccessclass), which is an immutable, cluster-scoped, custom resource providing a way for admins to specify policies that may be used to access buckets.
+- [BucketAccess](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1979-object-storage-support#bucketaccess), which is a cluster-scoped administrative custom resource which encapsulates fields from the `BucketAccessRequest (BAR)` and the `BucketAccessClass (BAC)`.
+
+All [COSI custom resource definitions](../crds) can be installed using the [kustomization](https://kustomize.io/) [file](../kustomization.yaml) and `kubectl` with following command:
+
+```sh
+kubectl create -k github.com/kubernetes-sigs/container-object-storage-api
```
Comment (YiannisGkoufas) [inline docs/deployment-guide.md:28]: There is a typo in this one, the word "interface" is missing. The correct one is:
```
kubectl create -k github.com/kubernetes-sigs/container-object-storage-interface-api
```

File: docs/deployment-guide.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+- Sidecar for the driver
+- Node Adapter
+
+### Quick Start
+
+Execute following commands to setup COSI:
+
+```sh
+# Install CRDs
+kubectl create -k github.com/kubernetes-sigs/container-object-storage-api
```
Comment (YiannisGkoufas) [inline docs/deployment-guide.md:23]: The correct one:
```
kubectl create -k github.com/kubernetes-sigs/container-object-storage-interface-api
```

**Review by wlan0 — APPROVED** [review]

**Pull request merged**